### PR TITLE
GG-32076: add property to disable statistics collection/usage.

### PR DESCRIPTION
**Pull request by Berkof**

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java [hunk 1]
```diff
@@ -437,11 +433,106 @@ public void dropColumnWhileNodeDown() throws Exception {
...
+        assertTrue(executeStatCfgCommands(ign1));
+    }
+
+    /**
+     * Run analyze/get/refresh/drop commands on specified node.
+     *
+     * @param ign Node to test.
+     * @return {@code true} if all commands pass successfully, {@code false} - otherwise.
+     */
+    private boolean executeStatCfgCommands(IgniteEx ign) throws IgniteInterruptedCheckedException {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java:495]: Types and methods names are NOT abbreviated except for well-known cases

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java [hunk 2]
```diff
@@ -437,11 +433,106 @@ public void dropColumnWhileNodeDown() throws Exception {
...
+        IgniteStatisticsManager statMgr = ign.context().query().getIndexing().statsManager();
+
+        int success = 0;
+        try {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java:499]: What for the try / catch and success couneter is used?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java [hunk 3]
```diff
@@ -437,11 +433,106 @@ public void dropColumnWhileNodeDown() throws Exception {
...
+            statMgr.collectStatistics(SMALL_TARGET);
+            success++;
+        } catch (Exception e) {
+            // NoOp.
+        }
+
+        if (GridTestUtils.waitForCondition(() -> statMgr.getLocalStatistics(SMALL_KEY) != null, TIMEOUT))
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java:506]: Just assertTrue?

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java [hunk 4]
```diff
@@ -318,7 +412,12 @@ public void refreshStatistics(List<StatisticsTarget> targets) {
...
     }
 
-    /** */
+    /**
+     *
+     *
+     * @param target
+     * @param tbl
+     */
     private void validate(StatisticsTarget target, GridH2Table tbl) {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java:421]: Javadoc is incomplete?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java [hunk 5]
```diff
@@ -437,11 +433,106 @@ public void dropColumnWhileNodeDown() throws Exception {
...
+
+        assertTrue(executeStatCfgCommands(ign0));
+        assertTrue(executeStatCfgCommands(ign1));
+
+        ign0statMgr.usageState(NO_UPDATE);
+
+        assertTrue(executeStatCfgCommands(ign0));
+        assertTrue(executeStatCfgCommands(ign1));
+
+        ign0statMgr.usageState(OFF);
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsConfigurationTest.java:473]: I guess we hav to check exception type/message in this case.

**Review by tledkov — COMMENTED** [review]
Should we check state at the 'dropAll' method?

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java [hunk 6]
```diff
@@ -124,6 +137,52 @@ else if (db == null)
...
+        StatisticsUsageState currState = usageState();
+        if (currState == ON || currState == NO_UPDATE)
+            enableOperations();
+    }
+
+    /**
+     * Enable statistics operations.
+     */
+    private void enableOperations() {
+
```
Comment (tledkov-gridgain) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java:174]: empty line

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java [hunk 7]
```diff
@@ -318,7 +412,12 @@ public void refreshStatistics(List<StatisticsTarget> targets) {
         }
     }
 
-    /** */
+    /**
+     *
+     *
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java:417]:
```suggestion
```

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java (hunk 6 above)
Comment (AMashenkov) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java:174]: Is it possible enableOperations() and disableOperations() is called out of order?
Does it make sense to synchronize them?

Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java:174]: Has sense - I'll add synchronized to both of them.

**Pull request merged**